prediction_market_agent/agents/known_outcome_agent/deploy.py

```diff
@@ -37,6 +38,12 @@ def pick_markets(self, markets: list[AgentMarket]) -> list[AgentMarket]:
             # been correctly bet on, and therefore the value of betting on them
             # is low.
             if not market_is_saturated(market=market):
+                if not has_question_event_happened_in_the_past(
+                    model=self.model, question=market.question
+                ):
+                    # event happened
+                    continue
```

Comment:
This is level 6 of indentation, which is quite error-prone for the future.

Why not just 

```python
if not market_is_saturated(market=market) and has_question_event_happened_in_the_past(
    model=self.model, question=market.question
):
    ...
```

?